chart: missing default `nodeSelector` value

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The default `nodeSelector` value in the chart should be `{}`, otherwise when one specifies their custom override - helm cannot type check it:

    coalesce.go:165: warning: skipped value for nodeSelector: Not a table.

**Expected behavior:**

It should be ok

**How to reproduce it (minimal and precise):**
<!-- Please let us know any circumstances for reproduction of your bug. -->

Run a chart with 

```yaml
nodeSelector:
  foo: bar
```
custom values file

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): ubuntu bionic
* Kernel (e.g. `uname -a`): `Linux <redacted> 4.15.0-96-generic #97-Ubuntu SMP Wed Apr 1 03:25:46 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud provider or hardware configuration: bare metal, nothing unique, just a commodity machine with ssds
* Rook version (use `rook version` inside of a Rook Pod): 1.3.0
* Storage backend version (e.g. for ceph do `ceph -v`): 14.2.8
* Kubernetes version (use `kubectl version`): 1.17.4
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): n/a
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

--- a/cluster/charts/rook-ceph/values.yaml
+++ b/cluster/charts/rook-ceph/values.yaml
@@ -16,7 +16,7 @@ resources:
     cpu: 100m
     memory: 256Mi
 
-nodeSelector:
+nodeSelector: {}
 # Constraint rook-ceph-operator Deployment to nodes with label `disktype: ssd`.
 # For more info, see https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
 #  disktype: ssd

